packetbeat/protos/mysql/mysql.go

```diff
@@ -140,6 +152,10 @@ type mysqlPlugin struct {
 	transactions       *common.Cache
 	transactionTimeout time.Duration
 
+	// prepare statements cache
+	prepareStatements       *common.Cache
+	prepareStatementTimeout time.Duration
```

Comment:
Prepared statements are accessed via address tuple. A mysql connection is allowed to only have this many prepared statements (applications requiring more or dynamic set of prepared statements tend to implement a Cache of active statements).

Why store the statements globally and time them out globally? I wonder if we want to move the statements to the TCP connection state.
Advantage is:  if the connection is gone all known prepared statements are gone as well
Disadvantage: if connection is dropped (internal timeout or parse error after gaps), then we loose the current state.

Alternatively bind to connections + timeout only if connection is dropped/errored:
- If TCP connection is properly closed we can can drop the state
- If connection is dropped we put the connection it's statements on timeout
- If new connection without TCP handshake is found we check if set of prepared statements is available for address pair and assign it to the current connection (stopping timeout).